Refonte de la page “À propos” : vraie description, Gandi & Café des Sciences, mises à jour, vraie mise en page

AmauryCarrade opened the pull request:
Suite à un mail de rappel de Gandi que nous avions demandé leur logo pour l'intégrer sur notre page _À propos_ (ils ne nous forcent pas, mais étaient étonnés que nous ne l'ayons pas fait alors que nous avions demandé de pouvoir le faire), j'ai voulu intégrer leur logo, donc.

Bon, ça a un peu dérapé (qui est surpris).

À voir la page, je me suis rappelé de la principale critique que je lui ai toujours fait, sans jamais prendre le temps de le corriger : ce n'est pas une page _À propos_, au mieux un [Colophon](https://indieweb.org/colophon). Et [j'aime beaucoup les colophons](https://amaury.carrade.eu/colophon), y'a [pas de souci](https://lu.sagebl.eu/colophon/), mais ce n'est pas ce qu'est attendu ici.

J'ai donc entreprit d'ajouter une vraie description. D'ajouter Gandi, et le Café des Sciences tant qu'on y est (il y a un logo prévu pour ça). Et puis ça manquait de certaines mentions légales. Et la liste des technologies n'était pas à jour. Et tout ça manquait un peu de style.

J'ai repris la description de multiples présentations (notamment celle au CNRS) et documents sur lesquels j'avais dû expliquer ce qu'est Zeste de Savoir.

Concernant la liste des technologies, la question se pose fatalement : que mettre, que ne pas mettre, tout mettre étant bien long ? J'ai fait un choix subjectif des technologies qui ont un usage/impact notable sur la plateforme, pour qu'on ait une vraie idée des outils utilisés pour construire Zeste de Savoir. J'ai également fusionné les deux listes, la séparation n'étant pas vraiment pertinente.

[**Cliquez ici pour obtenir un aperçu de la nouvelle page _À propos_** (à jour).](https://github.com/zestedesavoir/zds-site/assets/1417570/a4714b7c-b19b-4a6e-978d-3f63a26422a1)



## Questions à résoudre avant la QA


- [ ] Concernant les mentions légales, j'ai hésité. Il y a déjà des mentions légales dans les CGU (article 2) mais elles sont incomplètes. Il faut donc faire un choix :
    
    - déplacer les mentions légales hors des CGU (ce qui est assez classique) ; ou
    - remplacer les mentions légales de la page _À propos_ par un lien vers les mentions légales des CGU, et compléter les mentions légales dans les CGU (car il manque la ville de domiciliation de l'association, et _le ou la_ responsable de la publication — qui n'est pas l'association elle-même mais son ou sa représentante légale, i.e. son ou sa présidente).
    
- [ ] Secondement, l'adresse de l'association affichée est fausse (cf. commentaires plus bas). Faut-il mettre la bonne (`association@`) directement dans le fichier de paramètres de base ([zds/settings/abstract_base/zds.py](https://github.com/zestedesavoir/zds-site/blob/72978714c80daee8d0e3e4df6b653f68f795cb61/zds/settings/abstract_base/zds.py#L75-L81)), ou uniquement dans le fichier de production comme aujourd'hui ?
    
    Question subsidiaire, cette distinction a-t-elle le moindre intérêt ?
    Question sub-subsidiaire, est-ce _hors-scope_ de cette PR ?


## Contrôle qualité

  - Lancez `make build-front && make run-back` pour mettre à jour les styles et lancer le site ;
  - Rendez-vous sur [la page _À propos_](http://0.0.0.0:8000/pages/apropos/) et constatez que : 
      - le contenu est exact ;
      - il n'y a pas de fautes ;
      - les éléments montrés sont cohérents ; 
      - la page est visuellement propre, les styles ne sont pas cassés (sur écran large comme sur téléphone).



## Avant de fusionner

Je ferai une dernière passe typographique avant de fusionner, ne le faites pas immédiatement (toutes les apostrophes sont droites, là, notamment). Je le ferai à la fin au cas où des éléments doivent être ajoutés, afin de ne pas le faire vingt fois.

philippemilink committed:
Supprime le fichier généré errors/css/main.css du dépôt

Et intégre mieux sa construction au pipeline Gulp.

AmauryCarrade committed:
[about] Refonte de la page “à propos”.

J'étais parti pour juste insérer le logo Gandi, mais ça a dérapé. Oups.

- Restructuration de la page
- Ajout d'une vraie explication de ce qu'est Zeste de Savoir.
- Ajout de Gandi.
- Ajout du Café des Sciences.
- Remise en forme de la section présentant les technologies utilisées.
- Remise en forme de la section légale.

AmauryCarrade committed:
[about] Meilleures mentions légales.

- Ajout des mentions légales classiques sur la page à propos, à côté
  des autres.
- Doublon avec les CGU ? Mais idéalement, il faudrait au moins un
  report vers ces dernières, et les compléter, car il manque des
  informations — notamment, le siège social de l'association et son
  responsable de la publication. Un choix à faire.

AmauryCarrade committed:
[about] Ajout d'un lien vers les contributeurs et contributrices.

AmauryCarrade committed:
[about] Ajout de technologies manquantes.

jeanlapostolle commented:
J'aime beaucoup cette nouvelle page.

Petite coquille sur l'adresse mail dans les références légales et attributions qui est l'adresse gmail alors que ça devrait être "association@zestedesavoir.com"

AmauryCarrade commented:
Merci ! Et bien vu.

À noter que le problème existe en production car j'ai sourcé cette adresse depuis le même endroit que d'autres pages que je n'arrive à retrouver, à savoir, le ficher de configuration du site.

https://github.com/zestedesavoir/zds-site/blob/72978714c80daee8d0e3e4df6b653f68f795cb61/zds/settings/abstract_base/zds.py#L75-L81

D'ailleurs, cette adresse est dupliquée dans la configuration, faudrait vérifier que tout est bien utilisé.

Il faudrait aussi vérifier si cette adresse du CA est affichée quelque part, car je ne crois pas qu'elle soit utilisée en pratique (en tout cas à titre personnel je ne la monitore pas, je ne suis même pas sûr d'avoir accès).

AmauryCarrade commented:
Ah, c'était la page des CGU. Mais en fait le souci n’apparaît pas en prod, ou presque, en tout cas c'est pas la même adresse que ce que j'ai en local. Mais c'est `communication@`, alors qu'il faudrait plutôt `association@`.  Elle est modifiée dans le fichier de configuration de production (pourquoi ? pourquoi ne pas le mettre directement bon à la source ?).

https://github.com/zestedesavoir/zds-site/blob/72978714c80daee8d0e3e4df6b653f68f795cb61/zds/settings/prod.py#L149

AmauryCarrade committed:
[about] Version mobile.

AmauryCarrade committed:
[about] Lien vers la page de l'asso.

À terme, ces deux pages pourraient fusionner.

AmauryCarrade committed:
[about] Menus ajustements.

- Retour au serif (retiré par erreur suite à une refactorisation)
- Ajustement de marges

AmauryCarrade committed:
[about] Menus ajustements.

- Titres et sous-titres des blocs colorés plus petits sur mobile.

AmauryCarrade committed:
[about] Précisions sur Matomo (suivi côté serveur).

AmauryCarrade committed:
[oops] Annulation partielle d’une refactorisation automatique qu’a débordé.

AmauryCarrade committed:
[oops] Correction de l'indentation pour avoir un diff identité.

AmauryCarrade committed:
[oops] Moar style sous les technologies

Situphen commented:
Premièrement beau travail, le rendu est super !

Concernant les mentions légales, je pense qu'il faut suivre ce que font globalement tous les autres sites, c'est-à-dire les mettre dans une page à part des CGU avec un lien direct dans le footer.

Concernant l'adresse de l'association, ce n'est pas privé donc on peut simplement faire le changement dans le code de ZDS. Ceci dit, c'est le genre de paramètres qui selon moi ont toute leur place dans la configuration d'Ansible car cela permet de les modifier sans devoir publier une nouvelle version de ZDS.

J'ai parcouru très vite fait les modifications :

1. J'ai vu un commentaire en français dans le code SASS, est-ce volontaire ? Il me semble qu'actuellement la règle est de commenter le code en anglais, mais on pourrait décider d'autoriser le français dans les commentaires étant donné le peu de contributions anglophones dans l'histoire de ZDS. Peu m'importe personnellement. 
2. Tu as ajouté du code en SASS alors que tout le reste est en SCSS. Idem que le point précédent, est-ce qu'on souhaite autoriser le mélange des deux formats ? De manière idéaliste je préférerais un seul format mais j'imagine que tu dois préférer coder en SASS et tu es un des seuls contributeurs front, si ce n'est le seul, donc je ne m'y oppose pas.

Je ne sais pas si @philippemilink a un avis sur ces deux points. Mon objectif n'est pas de freiner cette PR mais surtout d'éviter un flou dans nos conventions/règles pour les prochaines PR.

AmauryCarrade commented:
Merci @Situphen !

> Concernant les mentions légales, je pense qu'il faut suivre ce que font globalement tous les autres sites, c'est-à-dire les mettre dans une page à part des CGU avec un lien direct dans le footer.

Tu ferais une page supplémentaire ? Tu penses que, supposant qu'on fasse effectivement comme à l'accoutumée ailleurs et qu'on les sépare des CGU, les mettre dans la page _À propos_ n'est pas suffisant ? Ou bien j'ai mal compris et tu parlais _justement_ de la-dite page ?

> Ceci dit, c'est le genre de paramètres qui selon moi ont toute leur place dans la configuration d'Ansible car cela permet de les modifier sans devoir publier une nouvelle version de ZDS.

Peut-être faudrait-il prévoir cette modification, alors ! Bien que ce ne soit pas très urgent. D'ici là, je peux modifier dans le dépôt, dans cette PR, si ça vous va.

> J'ai vu un commentaire en français dans le code SASS, est-ce volontaire ? Il me semble qu'actuellement la règle est de commenter le code en anglais, mais on pourrait décider d'autoriser le français dans les commentaires étant donné le peu de contributions anglophones dans l'histoire de ZDS. Peu m'importe personnellement.

Je plaide la mauvaise mémoire, j'avais souvenir qu'on avait le code en anglais mais les commentaires, documentations, commits et tickets/PR en français. Je vais traduire :) .

> Tu as ajouté du code en SASS alors que tout le reste est en SCSS. Idem que le point précédent, est-ce qu'on souhaite autoriser le mélange des deux formats ? De manière idéaliste je préférerais un seul format mais j'imagine que tu dois préférer coder en SASS et tu es un des seuls contributeurs front, si ce n'est le seul, donc je ne m'y oppose pas.

Clairement, mon confort de travail en Sass est _bien_ supérieur à celui en SCSS. Je me suis permis de l'utiliser, car : 
- Sass et SCSS sont conçus pour être mélangeables dans une même base de code ;
- Sass est très simple à comprendre si on connaît SCSS — à l'exception de la syntaxe courte pour les mixins (`+mobile`), que je peux m'efforcer de ne pas utiliser si cela vous semble être un compromis acceptable ;
- par extension, Sass est simple à écrire une fois qu'on a pigé le principe (pas de points-virgule et de l’indentation pour les niveaux — comme en Python, en fait !) ;
- enfin, Sass est, je trouve, plus lisible (code plus aéré, forcément bien indenté…).

Sass, c'est limite SCSS et Python qui ont fait un petit 🙃

Bref, je préfèrerais qu'on s'autorise à mélanger Sass et SCSS, vu ce qu'apporte Sass et la presque absence d'inconvénients que ce mélange a. D'ailleurs, pour ne rien vous cacher, j'avais prévu d'écrire Yuzu en Sass exclusivement.

AmauryCarrade committed:
[about/sass] Traduction des commentaires en anglais et ajout de PLEIN d'autres.

AmauryCarrade commented:
Je me suis laissé emporter par les commentaires. Bon, ils sont en anglais :d

AmauryCarrade committed:
[about/sass] Commentaire peu clair.

AmauryCarrade committed:
[about] Mauvaise version majeure de jQuery.

Ouais, c'était *à ce point* pas à jour.

AmauryCarrade committed:
[about] Sélection invisible dans le bloc des technologies.

Il fallait l'ajouter à la liste des éléments avec sélection inversée.

qwertygc commented:
Pour le texte, s'inspirer aussi de https://zestedesavoir.com/forums/sujet/17025/auteurs-externes-faire-venir-des-universitaires/

AmauryCarrade commented:
Ah oui bien vu, il y a sans doutes des choses à reprendre :)

AmauryCarrade committed:
[numérotation] Factorisation du SCSS ; style.

- Inutile de répéter les sélecteurs, on fait du SCSS, on peut imbriquer.
- Correction du style de code d'après les règles editorconfig : indentation avec deux espaces.

AmauryCarrade committed:
[numérotation] Correction des soucis d'espacement.

- On ne s'appuie plus sur des espaces dans le contenu, mais sur une marge à la
  taille relative à la largeur d'une lettre de la police courante.
- Les compteurs sont réinitialisés en une ligne (plus clair, inutile de
  s’embêter).

AmauryCarrade committed:
[numérotation] Tentative de mise en avant du titre

- Les numérotations sont à peine plus claires.
- J'ai tenté aussi de retirer la graisse. Je ne sais que choisir.

AmauryCarrade committed:
[about] Test Kitchen manquant et formulation imprécise.

AmauryCarrade committed:
[about] Cohérence de la ponctuation

Arnaud-D commented:
Petit commentaire sur le contenu :

* La page "À propos" que tu es en train de faire devrait intégrer la page "Association" (peu importe le nom), cette dernière disparaissant alors.
* On a choisi de renommer le "À propos" en "Crédits techniques" (voir #6515) sans en changer le contenu. Il faudra gérer ça au rebase. Je pense qu'il faut garder cette page séparée. Et pas besoin d'en faire des caisses non plus, parce que le dépôt est aussi là pour ça ; en l'état je trouve que c'est trop.

En résumé : d'un côté une page "à propos" avec tes nouvelles infos et celles de la page "association" (description de l'asso et du site) et d'un autre côté un colophon. Ça mérite d'être séparé selon moi.

Sinon, je pense aussi qu'il faudrait une page "mentions légales" dédiée avec lien direct dans le pied de page.

Sympa le rendu graphique sinon. :)

AmauryCarrade commented:
> La page "À propos" que tu es en train de faire devrait intégrer la page "Association" (peu importe le nom), cette dernière disparaissant alors.

Plus j'y pense, plus je pense également que ce serait une bonne idée, et c'était dans mes projets. Je réfléchissais juste comment l'intégrer (tant éditorialement que graphiquement). Pour l'édito, le texte cité par @qwertygc est une base intéressante. De manière plus générale, je suis ouvert à toutes les idées !

> On a choisi de renommer le "À propos" en "Crédits techniques" (voir https://github.com/zestedesavoir/zds-site/pull/6515) sans en changer le contenu. Il faudra gérer ça au rebase. Je pense qu'il faut garder cette page séparée.

Je n'avais pas vu cette PR, mais en effet, ce n'est pas une mauvaise idée. Je suis assez neutre quant au fait de séparer ou non, et je comprends les arguments pour séparer, moi ça me va bien. Est-ce que tu sais à quel point la PR #6515 est sur le point d'être fusionnée ? Sinon, je peux la reprendre directement dans cette PR, ça pourrait peut-être être plus simple.

> Et pass besoin d'en faire des caisses non plus, parce que le dépôt est aussi là pour ça ; en l'état je trouve que c'est trop.

Je ne suis pas trop d'accord : j'ai pas mal limité pour que ça reste une vue d'ensemble, mais je trouve que c'est intéressant de mettre l'important dans un peu toutes les catégories. J'y vois un but pédagogique, étant donné que le site enseigne (notamment) l'informatique, montrer tout ce qu'on finit par utiliser pour faire une plateforme web un peu complexe est intéressant, je trouve.

J'aurais bien plus compris l'envie de diminuer si c'était resté en page unique, pour être honnête, car il est vrai qu'en l'état, ça prend pas mal de place. Mais sur une page dédiée… je ne vois pas le souci (sinon celui de la maintenance, mais j'ai volontairement choisi des technologies qu'on utilise de façon stable : la seule qu'il faudra changer à court ou moyen terme, sauf grosse surprise, c'est ElasticSearch).

Cela dit, je reste bien évidemment ouvert à la discussion — qu'est-ce que tu trouves superflu ?

> Sympa le rendu graphique sinon. :)

Merci !

Arnaud-D commented:
> Est-ce que tu sais à quel point la PR https://github.com/zestedesavoir/zds-site/pull/6515 est sur le point d'être fusionnée ?

À deux doigts d'être fusionnée, vraiment. Il reste la QA à faire (que tu peux faire si tu veux !), mais c'est juste pour vérifier la prise en compte de commenaires purement techniques.

Pour moi, elle doit être fusionnée séparément. Ça sera toujours des choses à vérifier en moins dans la présente PR.

> Cela dit, je reste bien évidemment ouvert à la discussion — qu'est-ce que tu trouves superflu ?

C'est plus que je trouve le concept de lister beaucoup de choses un peu overkill, pas spécialement des choses spécifiques.

Mais ça m'est un peu égal que ça soit long, ça ne me dérange pas.

Par contre, et c'est important pour la maintenance : ne pas citer de version spécifiques (à part Python peut-être), parce qu'elles changent régulièrement et on n'a pas envie d'avoir une page obsolète. Par exemple, Django 3, dans quelques mois, ça ne sera plus vrai et on oubliera de changer la page. :-°

AmauryCarrade commented:
> À deux doigts d'être fusionnée, vraiment. Il reste la QA à faire (que tu peux faire si tu veux !), mais c'est juste pour vérifier la prise en compte de commenaires purement techniques.

Je ne garantie pas d'avoir le temps mais je vais voir, peut-être :) Et sinon ok !

> Pour moi, elle doit être fusionnée séparément. Ça sera toujours des choses à vérifier en moins dans la présente PR.

Fair. Le rebase risque juste d'être un peu pénible (fichier renommé mais en fait il ne faut pas qu'il le soit, etc.), cela dit, rien d'insurmontable.

> Par contre, et c'est important pour la maintenance : ne pas citer de version spécifiques (à part Python peut-être), parce qu'elles changent régulièrement et on n'a pas envie d'avoir une page obsolète. Par exemple, Django 3, dans quelques mois, ça ne sera plus vrai et on oubliera de changer la page. :-°

Sur ce point précis, si ça peut te rassurer, les premiers concepteurs de cette page ont prévu le coup : 

https://github.com/zestedesavoir/zds-site/blob/ae460de826eb996b2f4becc8995b24fe421a1d9b/zds/pages/views.py#L62-L74

Il n'y a que jQuery qui pourrait perdre sa précision de version — d'ailleurs j'ai vu que ça a été fait dans la PR que tu citais. De toute façon, c'est pas comme si on était en train de s'en débarrasser petit à petit.

philippemilink commented:
Après en avoir discuté en [réunion des dev's](https://zestedesavoir.com/forums/sujet/11108/reunion-des-dev-de-zeste-de-savoir/?page=7#p252794), voici les points qui ont été décidés : 

> Après discussion, on a besoin des pages suivantes :
> - "L’association" : présentation de l’association (comme décidé lors de la dernière AG) et lien d’adhésion
> - "Technologies" : présentation des technologies utilisées et lien vers la documentation de l’API
> - "Mentions légales" : mentions légales, CGU, cookies, licences
>
> On supprime du pied de page les liens "API", "Cookies" et "Adhérer à l’association".
>
> Concernant la page décrivant les technologies utilisées, on souhaite mettre du détail, puisque ça peut intéresser les visiteurs et visiteuses qui consultent après tout un site de cours (à majorité informatique). Par rapport à ce qui est proposé dans la PR, on ajoute plus de texte, pour expliquer le rôle des différentes technologies, pourquoi on en a besoin. On peut également dire qu’on accueille les développeurs débutant et mettre un lien vers les tickets faciles.
>
> Amaury a besoin d’aide pour ce qu’il y a à mettre dans la page qui présente l’association. Ces éléments peuvent venir du rapport d’activité (dont il faudra mettre le lien sur cette page).

qwertygc commented:
Du coup, ce qu'on doit faire : 

- Enlever la partie technique (page dédiée)
- Mettre le texte ici : https://zestedesavoir.com/forums/sujet/17029/reecriture-de-la-page-association/?page=1#p251625

Situphen commented:
J'ai rebasé la branche et je l'ai mise sur la bêta, sans modification pour le moment.

AmauryCarrade committed:
Revert "[numérotation] Factorisation du SCSS ; style."

This reverts commit 316f604eb32416d1498ac937d5e6e2c8a20390f6.

AmauryCarrade committed:
Revert "[numérotation] Correction des soucis d'espacement."

This reverts commit edf7b3532cd753e75214b6a2c44e68aa3c3ff094.

AmauryCarrade committed:
Revert "[numérotation] Tentative de mise en avant du titre"

This reverts commit aabf5bf938f63183492aa89673f0990bdc99aae1.

AmauryCarrade committed:
Reprise des pages “à propos” — Technologies

- Division de la page en deux : une partie pour les
  technologies (un colophon), et une pour un vrai
  “à propos” et pour l’association.
- Rédaction de la page “technologies”.

AmauryCarrade commented:
Bon ; j'ai commencé la reprise de cette PR. Pour le moment, la page des technologies me semble correcte. Il me reste à intégrer la page “à propos”, que je vais fusionner avec la page “Association” — à moins que quelqu’un n’y voit un problème !

Arnaud-D commented:
J'ai regardé et j'ai trouvé ça bien ! Pas de problèmes avec la fusion des deux pages, j'avais proposé ça auparavant et ça me paraît toujours une bonne logique. :)

AmauryCarrade commented:
Il faudrait ajouter un lien vers le [code de conduite](https://zestedesavoir.com/forums/sujet/17500/code-de-conduite-de-la-communaute/) dans la page _à propos_.

Situphen commented:
Bonsoir,
Quelqu'un peut-il récapituler où nous en sommes ici ?

AmauryCarrade commented:
@Situphen Le [message du 24 juin](https://github.com/zestedesavoir/zds-site/pull/6527#issuecomment-2187332319) est toujours d’actualité, n’ayant pas pu avancer depuis.
